internal/metrics/metrics.go

```diff
@@ -1,22 +1,60 @@
 package metrics
 
 import (
+	"context"
+	"fmt"
+
+	routev1 "github.com/openshift/api/route/v1"
+	"k8s.io/apimachinery/pkg/types"
 	addonapiv1alpha1 "open-cluster-management.io/api/addon/v1alpha1"
 	clusterv1 "open-cluster-management.io/api/cluster/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 type MetricsValues struct {
 	Enabled bool `json:"enabled"`
+	// TODO: revert this hack to the official way as recommended by the docs.
+	// See https://open-cluster-management.io/developer-guides/addon/#values-definition.
+	AddonInstallNamespace string `json:"addonInstallNamespace"`
+	DestinationEndpoint   string `json:"destinationEndpoint"`
 }
 
-func GetValuesFunc(_ client.Client, _ *clusterv1.ManagedCluster, _ *addonapiv1alpha1.ManagedClusterAddOn) (MetricsValues, error) {
+func GetValuesFunc(
+	k8sClient client.Client,
+	_ *clusterv1.ManagedCluster,
+	mca *addonapiv1alpha1.ManagedClusterAddOn,
+	adoc *addonapiv1alpha1.AddOnDeploymentConfig,
+) (MetricsValues, error) {
+	endpoint, err := getDestinationEndpoint(k8sClient, adoc)
+	if err != nil {
+		return MetricsValues{}, fmt.Errorf("failed to get metrics destination endpoint: %w", err)
+	}
 	values := MetricsValues{
-		Enabled: false,
+		Enabled:               true,
+		AddonInstallNamespace: mca.Spec.InstallNamespace,
+		DestinationEndpoint:   endpoint,
 	}
+	return values, nil
+}
 
-	// Get necessary values
+func getDestinationEndpoint(k8sClient client.Client, adoc *addonapiv1alpha1.AddOnDeploymentConfig) (string, error) {
+	if adoc == nil {
+		return "", nil
+	}
```

Comment:
```suggestion
```

Reply:
Related to https://github.com/rhobs/multicluster-observability-addon/pull/14#discussion_r1467534270